hack/deploy-metrics-adapter.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
+function usage() {
+  echo "This script will deploy karmada-metrics-adapter on host cluster"
+  echo "Usage: hack/deploy-metrics-adapter.sh  <HOST_CLUSTER_KUBECONFIG> <HOST_CLUSTER_NAME>>"
+  echo "Example: hack/deploy-metrics-adapter.sh ~/.kube/config karmada-host"
+}
+
+if [[ $# -ne 2 ]]; then
+  usage
+  exit 1
+fi
+
+# check kube config file existence
+if [[ ! -f "${1}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${1}', not existed.\n"
+  usage
+  exit 1
+fi
+HOST_CLUSTER_KUBECONFIG=$1
+
+# check context existence
+if ! kubectl config get-contexts "${2}" --kubeconfig="${HOST_CLUSTER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${2}' not in ${HOST_CLUSTER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+HOST_CLUSTER_NAME=$2
+
+# install metrics adapter to host cluster
+if [ -n "${KUBECONFIG+x}" ];then
+  CURR_KUBECONFIG=$KUBECONFIG # backup current kubeconfig
+fi
+
+export KUBECONFIG=$HOST_CLUSTER_KUBECONFIG
+echo "using kubeconfig: "$KUBECONFIG
+
+# deploy karmada-metrics-adapter
+kubectl --context="${HOST_CLUSTER_NAME}" apply -f "${REPO_ROOT}/artifacts/deploy/karmada-metrics-adapter.yaml"
+
+# make sure that karmada-metrics-adapter is ready
+util::wait_pod_ready "${HOST_CLUSTER_NAME}" "${KARMADA_METRICS_ADAPTER_LABEL}" "${KARMADA_SYSTEM_NAMESPACE}"
+
+# deploy karmada-metrics-adapter-apiservice
+kubectl --context="karmada-apiservice" apply -f "${REPO_ROOT}/artifacts/deploy/karmada-metrics-adaptor-apiservice.yaml"
```

Comment:
We are uncertain whether the context name should be `karmada-apiserver`. This parameter should be specified as an input flag.

Reply:
The logic behind taking karmada-apiserver as the context was that Karmada is using two different contexts i.e. karmada-apiserver and karmada-host that seem to be referring to the same entity. All the Kubectl request needs to be passed to the cluster where the Karmada control plane is hosted and aggregation should be with the API server of that cluster. In this case, to me, it seems both the karmada-apiserver and karmada-host are referring to the same entity. I took context as karmada-apiserver as it looked more of a convention to me as it was being used by both the API service for aggregation purposes. I wasn't sure either whether it is being used for security purposes or something else, so went with the convention.

Reply:
There seems to be a small typo, "karmada-apiservice" -> "karmada-apiserver". I will fix it.

Reply:
Karmada is installed is a specific cluster(host cluster), it contains `kube-apiserver` component, named `karmada-apiserver`. So karmada-host is used to connect to the host cluster's kube-apiserver, karmada-apiserver is used to connect to `karmada-apiserver`. They are different entity.

Reply:
I am still a bit confused, as it is a part of the same cluster kube-apiserver itself is capable of handling all such requests easily. Then why we are using two contexts here? I am not sure whether any permission boundaries exist or need to exist in the case of aggregated API.

Reply:
Karmada is responsible for workload propagation, if we use the clusters' apiserver, the workload will be created in the cluster, but what we want is: the deployment should be created in member clusters. So we have to create ns, deploy another K8s there, and cut some functions for not creating real workloads there, just sync it to member clusters with policy.